examples/basic-infection/README.md

```diff
@@ -183,82 +186,130 @@ infection status:
-```
-init(context) {
-    context.observe_person_property_event(handle_person_property_change(context, person_id, previous_infection_status));
+fn init(context) {
+    context.add_report::<IncidenceReport>("incidence");
+    context.observe_person_property_event::<InfectionStatus>(handle_infection_status_change);
 }
 ```
 
-The method `handle_person_property_change` writes a new line to the report file with the change in the person property of infection status.
-```
-fn handle_person_property_change(context, person_id, infection_status){
-    report_data = (t = context.get_time(), person_property_type = Type(infection_status), person_property_value = context.get_infection_status(person_id), person_id = person_id);
-    context.print_report_data(report_data); // Method to print to csv, tsv to be implemented in the
+The method `handle_infection_status_change` writes a new line to the report file, recording the status and the current time:
+
+```rust
+fn handle_infection_status_change(context, person_id, prev, current) {
+    context.add_report(Incidence(
+        person_id,
+        infection_status: current,
+        t: context.get_current_time()
+    ));
 }
 ```
 
+One consideration here is that if the callback references context, it should
+provide the state of the context exactly at the time the event was released.
+
 #### Periodic Reports
-To report the current state of each Infection Status (Susceptible, Infected, or Recovered), this report needs an additional data structure that keeps a count of individuals on each state and is updated every time an event is released due to changes in infection status.
+The second type of report records something about the current state of the
+simulation at the end of a period, such as after every day.
+
+For this example, we record a count of the number of individuals with each
+infection status (S,I,R) at the end of every day:
+
 ```
-person_property_counter = struct {
-    map: hashmap<PropertyType, hashmap<property_value, int>>;
-    // e.g., this map could represent 100 susceptible people
-    // <InfectionStatus, <Susceptible, 100>>
-}
-context.add_data_container(person_property_counter)
+day,infection_status,count
+0,Suceptible,92
+0,Infected,8
+0,Recovered,0
+1,Suceptible,89,
+1,Infected,12
+1,Recovered,0
+...
 ```
-At initialization, the property counter is initialized with the current state of all the population. The report also observes changes in person properties which are handled by a function that updates counts in the property counter. The first report day is scheduled for t = 0.
+
+In this case, we could actually compute each daily summary in
+post-processing from the instantaneous reports instead of generating a second
+set of periodic reports. However, we want a summary of properties
+which are not otherwise recorded, such as perhaps whether an individual is
+hospitalized.
```

Comment:
Yes, currently hospitalized people would be great. I also think maybe we want to know how many susceptible people are at some point. If we have a model where there are births and deaths, it might be difficult to keep track of the actual number of susceptible individuals in post-processing, and maybe more difficult if we include categories (e.g., age).